lib/plugins/aws/deploy/index.js

```diff
@@ -22,41 +28,30 @@ class AwsDeploy {
     Object.assign(
       this,
       validate,
+      runPackage,
+      extendedValidate,
       createStack,
-      generateArtifactDirectoryName,
-      mergeCustomProviderResources,
       setBucketName,
       cleanupS3Bucket,
       uploadArtifacts,
       updateStack,
-      monitorStack,
-      configureStack,
-      mergeIamTemplates
+      monitorStack
     );
 
     this.hooks = {
-      'before:deploy:initialize': () => BbPromise.bind(this)
-        .then(this.validate),
-
       'deploy:initialize': () => BbPromise.bind(this)
-        .then(this.configureStack),
-
-      'deploy:setupProviderConfiguration': () => BbPromise.bind(this)
-        .then(this.createStack)
-        .then(this.mergeIamTemplates),
-
-      'before:deploy:compileFunctions': () => BbPromise.bind(this)
-        .then(this.generateArtifactDirectoryName),
+        .then(this.validate)
+        .then(this.runPackage)
+        .then(this.extendedValidate),
 
       'deploy:deploy': () => BbPromise.bind(this)
-        .then(this.mergeCustomProviderResources)
+        .then(this.createStack)
```

Comment:
This could be combined with the entry point PR to have the deploy split up in more fine-grained hookable lifecycle events.